superset/db_engine_specs.py

```diff
@@ -1089,42 +1098,39 @@ def _get_fields(cls, cols: List[dict]) -> List[ColumnClause]:
         return column_clauses
 
     @classmethod
-    def _filter_presto_cols(cls, cols: List[dict]) -> List[dict]:
+    def _filter_out_array_nested_cols(
+            cls, cols: List[dict]) -> Tuple[List[dict], List[dict]]:
         """
-        We want to filter out columns that correspond to array content because expanding
-        arrays would require us to use unnest and join. This can lead to a large,
-        complicated, and slow query.
-
-        Example: select array_content
-                 from TABLE
-                 cross join UNNEST(array_column) as t(array_content);
-
-        We know which columns to skip because cols is a list provided to us in a specific
-        order where a structural column is positioned right before its content.
+        Filter out columns that correspond to array content. We know which columns to
+        skip because cols is a list provided to us in a specific order where a structural
+        column is positioned right before its content.
 
         Example: Column Name: ColA, Column Data Type: array(row(nest_obj int))
                  cols = [ ..., ColA, ColA.nest_obj, ... ]
 
         When we run across an array, check if subsequent column names start with the
         array name and skip them.
         :param cols: columns
-        :return: filtered list of columns
+        :return: filtered list of columns and list of array columns and its nested fields
         """
         filtered_cols = []
+        array_cols = []
         curr_array_col_name = ''
```

Comment:
Nit: I'd initialize this to `None`, to make it clear that is hasn't been set yet. I know you check the truth value on line 1121 (1117 in the old file) and this works, but explicit is better than implicit.